app/utils.py

```diff
@@ -150,10 +150,71 @@ def list_to_string(lst):
             res += c
 
     return res
+	
+def get_user_restaurants(userId):
+	"""
+    Queries our database for all the restaurants a user has
+    """
+	# restaurants = models.Restaurant.query(Restaurant).join(Restaurant.id==UserRestaurant.restaurant_id).filter_by(UserRestaurant.user_id==userId).all()
+	
+	restaurantNames = []
+	userRestaurants = models.UserRestaurant.query.filter_by(user_id=userId).all()
+	
+	for userRestaurant in userRestaurants:
+		restaurant = models.Restaurant.query.filter_by(id=userRestaurant.restaurant_id).first()
+		
+		if not restaurant.name in restaurantNames:
+			restaurantNames.append(restaurant.name)
+		
+	return restaurantNames
+
+def get_restaurant_details_by_name(restaurantName, locId, start):
```

Comment:
this is not performant; it takes about 3 seconds to pull back a single restaurant. Instead, this should work like get_restaurant_details(). I think you should also see if you can accomplish this same thing with get_restaurant_details as opposed to individually searching each restaurant one at a time